### PR TITLE
Merge upstream up to 425d77f8124a50fc033e8fb3bdf7b89a6a25f4fa

### PR DESCRIPTION
**Pull request by ClearlyClaire**
As explained in https://github.com/glitch-soc/mastodon/pull/2363#issuecomment-1672641370, I find it easier to do the merge/port PRs myself. That being said, thanks @Plastikmensch for #2363, this has been (and will continue to be) helpful for triaging commits to port/skip/postpone

## Conflicts

- `.rubocop_todo.yml`:
  Upstream regenerated this file, glitch-soc had a specific ignore.
- `README.md`:
  Upstream updated its README, but glitch-soc has a completely different one.
  Kept glitch-soc's README

### Postponed changes:

- d4807a5e645b9363c7f8d6702186df4325bc5a97 (link preview change, see #2343)
- 4c999a736c09223418749b85304b128e746be64d (link preview change, see #2343)
- 01f0cffc2c5b14e0a7932539896783cc11b2a052 (link preview change, see #2343)
- 2cbdff97cedcf87f567effa782e0314c849334c6 (role badges redesign, see #2341)
- 6308dca76aef6dd8265813d58f7aab3f96eb21cc (UI focus change, see #2342)

## Ported changes

- 71fd70335adecfc53c3dd2d57e1483ff60adb530

## Skipped changes

- 4d3a12921254f992f649ad84f88ef947b9adc13d (already fixed in #2336)

### CODE REVIEW

File: app/controllers/mail_subscriptions_controller.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@
   before_action :set_user
   before_action :set_type
 
+  protect_from_forgery with: :null_session
```
[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply app/controllers/mail_subscriptions_controller.rb:12]: As noted upstream, we want users to be able to unsubscribe from their email client, which does a `POST` request.

**Comment by Plastikmensch** [comment]
> As explained in [#2363 (comment)](https://github.com/glitch-soc/mastodon/pull/2363#issuecomment-1672641370), I find it easier to do the merge/port PRs myself. That being said, thanks @Plastikmensch for #2363, this has been (and will continue to be) helpful for triaging commits to port/skip/postpone

Glad it was helpful :3
I could maintain a list (in an issue or so) of upstream commits which could be ported and which need to be postponed, if that's something that would help you.

One clarification question though: Does "port PRs" also mean porting already postponed changes?

**Comment by ClearlyClaire** [comment]
>  I could maintain a list (in an issue or so) of upstream commits which could be ported and which need to be postponed, if that's something that would help you.

It is useful! It's also very useful to point commits to already-open issues!

> One clarification question though: Does "port PRs" also mean porting already postponed changes?

Yeah also, although it's a little more self-contained and so a little easier to maintain.

**Comment by Plastikmensch** [comment]
> It is useful! It's also very useful to point commits to already-open issues!

Can do!
I'm a bit unsure if I should create one issue and keep updating it, or create a new one after every upstream merge.
(The latter would be a bit less spammy with references and keep the backreferences useful)

> Yeah also, although it's a little more self-contained and so a little easier to maintain.

Okay, then I'll probably stop doing those.

**Pull request merged**